src/ORM/Filters/SearchFilter.php

```diff
@@ -309,6 +309,32 @@ public function getDbName()
         return '"' . implode('"."', $parts) . '"';
     }
 
+    /**
+     * Return getDbName() and ensures that the passed query has all the joins necessary to reference it.
+     *
+     * @param $query The DataQuery to add joins to
+     * @return string A table-qualified, quoted field name
+     */
+    public function getDbNameAndAddToQuery(DataQuery $query)
```

Comment:
This name is a mouthful, it was intended to clarify that it both modifying the query and then returning a reference to that field on the modified query.

Another approach to this would be to refactor this to live more on the DataQuery object itself, e.g.

```
$query->provideAccessToField($this->name)
```

```
$query->provideAccessToAggregate($this->aggregate['function'], $this->aggregate['column'])
```

This might also mean that joinSubclassTableForField doesn't need to be added to the public API anymore. It would mean that the SearchFilters need more refactoring, and it might mean that we want to formally deprecate getDbName() as well.

Reply:
Everything I've added has been overly verbose, so it stays in character. :)

Reply:
What do you think about shifting more of this responsibility from SearchFilter to DataQuery?